src/main/java/org/springframework/data/hazelcast/HazelcastKeyValueAdapter.java

```diff
@@ -38,13 +39,12 @@
 	private HazelcastInstance hzInstance;
 
 	public HazelcastKeyValueAdapter() {
-		this(Hazelcast.newHazelcastInstance());
+		super(new HazelcastQueryEngine());
+		Config cfg = new Config(Constants.HAZELCAST_INSTANCE_NAME);
+		this.hzInstance = Hazelcast.getOrCreateHazelcastInstance(cfg);
```

Comment:
It's now using an existing instance instead of creating a new one. I'm not sure if there is a use case starting two instances in the same JVM but if there is one, then this will change the existing behavior.